app/src/main/java/vn/sunasterisk/music_67/ui/search/SearchAdapter.kt

```diff
@@ -0,0 +1,74 @@
+package vn.sunasterisk.music_67.ui.search
+
+import android.content.Context
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.bumptech.glide.Glide
+import kotlinx.android.synthetic.main.item_track.view.*
+import vn.sunasterisk.music_67.R
+import vn.sunasterisk.music_67.data.model.Track
+import vn.sunasterisk.music_67.utils.TimeConvert
+
+class SearchAdapter(val context: Context,
+                    private var tracks: ArrayList<Track>,
+                    var trackClickListener: TrackClickListener
+) : RecyclerView.Adapter<SearchAdapter.ViewHolder>() {
+    private val inflater: LayoutInflater
+
+    init {
+        inflater = LayoutInflater.from(context)
+    }
+
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+        val view = inflater.inflate(R.layout.item_track, viewGroup, false)
+        return ViewHolder(context, view, trackClickListener)
+    }
+
+    override fun getItemCount(): Int {
+        return tracks.size
+    }
+
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        viewHolder.bindView(tracks[position])
+    }
+
+    class ViewHolder(
+        val context: Context,
+        trackView: View,
+        val trackClickListener: TrackClickListener
+    ) : RecyclerView.ViewHolder(trackView) {
+        init {
+            trackView.setOnClickListener {
+                trackClickListener.onTrackClicked(adapterPosition)
+            }
+            trackView.imageMore.setOnClickListener {
+                trackClickListener.onMoreClicked(adapterPosition)
```

Comment:
để ViewHolder implements rồi setOnClickListener(this)

Reply:
đoạn này em không hiểu ý anh lắm

Reply:
verbal explanation =))